### PR TITLE
Add more URLs to setup.py

### PR DESCRIPTION
**Pull request by mikemhenry**
## Description
Added some relevant project URLs to `setup.py`

## Motivation and Context
We should make it as easy as possible for users to find all of the URLs associated with the project. These changes will make it easy for someone to find these URLs as they will be listed on [here](https://pypi.org/project/signac/) under project links.

## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

Example for a changelog entry: `Fix issue with launching rockets to the moon (#101, #212).`

### CODE REVIEW

**Comment by mikemhenry** [comment]
If we are happy with these additions, they can be propagated to the other signac repositories.

**Comment by csadorf** [comment]
Generally a good idea, I'm just wondering what happens if we decide to move the source code for example? The docs have a pretty stable URL, but I'm not so sure about the source code long-term.

**Comment by mikemhenry** [comment]
If we move where we have the source code, we will update the link. Just as an example what this looks like, check [`pandas`](https://pypi.org/project/pandas/) and note that they have links on the left hand side under "Project links". Right now [`signac`](https://pypi.org/project/signac/) is missing a few of those links and I think that impacts things like [libraries.io](https://libraries.io/pypi/signac) which show things like the repository still on bitbucket. 

It will be something we need to remember to update, but it will show up with a `grep` of the codebase when we migrate and need to update links.

File: setup.py [hunk 1]
```diff
@@ -55,6 +55,12 @@
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
+    
+    project_urls={
+    'Documentation': 'https://docs.signac.io',
+    'Source': 'https://github.com/glotzerlab/signac',
+    'Tracker': 'https://github.com/glotzerlab/signac/issues',
```
Comment (bdice) [inline setup.py:62]:
```suggestion
        'Homepage': 'https://signac.io'
        'Documentation': 'https://docs.signac.io',
        'Source Code': 'https://github.com/glotzerlab/signac',
        'Issue Tracker': 'https://github.com/glotzerlab/signac/issues',
```

File: setup.py [hunk 2]
```diff
@@ -55,6 +55,13 @@
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
+    
+    project_urls={
+        'Homepage': 'https://signac.io'
```
Comment (csadorf) [inline setup.py:60]:
```suggestion
        'Homepage': 'https://signac.io',
```
This is why the CI pre-checks are failing.

Comment (bdice) [reply setup.py:60]: My mistake, I was fixing other pre-checks. 😆

File: setup.py [hunk 3]
```diff
@@ -55,6 +55,13 @@
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
+
```
Comment (bdice) [inline setup.py:58]:
```suggestion

```

File: setup.py [hunk 4]
```diff
@@ -55,6 +55,12 @@
         "Programming Language :: Python :: Implementation :: CPython",
         "Programming Language :: Python :: Implementation :: PyPy",
     ],
+    project_urls={
```
Comment (bdice) [inline setup.py:58]:
```suggestion

    project_urls={
```

**Comment by mikemhenry** [comment]
@vishav1771 Let me know if you have any questions about this PR :)

**Comment by vishav1771** [comment]
LGTM :)

**Pull request merged**